orchestration/dagster_orchestration/hca_orchestration/contrib/retry.py

```diff
@@ -0,0 +1,49 @@
+import time
+from typing import Any, Callable, TypeVar
+
+
+class RetryException(Exception):
```

Comment:
nit: inherit from `RuntimeError` for user-defined exceptions, inheriting from `Exception` implies a system-level failure